samtranslator/model/sam_resources.py

```diff
@@ -2237,6 +2245,93 @@ def _construct_appsync_api_resources(
 
         return api, cloudwatch_role
 
+    def _parse_auth_properties(self, api: GraphQLApi, auths: List[aws_serverless_graphqlapi.Auth]) -> None:
+        """
+        Parse the Auth properties in a Serverless::GraphQLApi resource.
+
+        This function does not return a value. The GraphQLApi object is modified directly so that we don't
+        return multiple config properties which are mostly None values.
+        """
+        if not auths:
+            raise InvalidResourceException(self.logical_id, "'Auth' must contain at least one valid authorizer.")
+
+        additional_auths: List[AdditionalAuthenticationProviderType] = []
+
+        # In each Auth index, you should only define a max of two properties. The "Type" property, and if
+        # necessary the associated config as well.
+        MAX_AUTH_PROPERTIES = 2
+
+        for index, auth in enumerate(auths):
```

Comment:
Nit: You could reduce the complexity of this method by pulling lines 2266-2272 into a method (it's also not clear why `_validate_auth_type_and_config` doesn't validate that you haven't defined too many properties); then you can treat the first element separately and have the loop for additional providers be over the rest of the list. 
This is, imo, clearer than having a special case for the first element of the list and a continue.

Reply:
Great points, not really sure why I separated the too many properties check either from the validation method... With our slightly altered Auth design, I will also be treating the first element separately which should make it a lot clearer to read. Thank you!